src/contracts/core/DelegationManager.sol

```diff
@@ -785,6 +1018,146 @@ contract DelegationManager is
+            emit StrategyWithdrawalDelaySet(strategy, prevStrategyWithdrawalDelay, newStrategyWithdrawalDelay);
+        }
+    }
+
+    /**
+     *
+     *                         SLASHING AND SHARES HELPER FUNCTIONS
+     *
+     */
+
+    /**
+     * @notice helper pure to calculate the scaledShares to store in queue withdrawal
+     * and the shares to decrement from StrategyManager/EigenPodManager
+     */
+    function _calculateSharesToQueueWithdraw(
+        uint256 sharesToWithdraw,
+        uint256 stakerScalingFactor,
+        uint64 totalMagnitude
+    ) internal pure returns (uint256 sharesToDecrement, uint256 scaledStakerShares) {
+        // TODO: DOUBLE CHECK THIS BEHAVIOR
+        // NOTE that to prevent numerator overflow, the max sharesToWithdraw is
+        // x*1e36 <= 2^256-1
+        // => x <= 1.1579e41
+        sharesToDecrement =
+            (sharesToWithdraw * SlashingConstants.PRECISION_FACTOR_SQUARED) / (stakerScalingFactor * totalMagnitude);
+        scaledStakerShares = sharesToWithdraw * stakerScalingFactor / SlashingConstants.PRECISION_FACTOR;
+        return (sharesToDecrement, scaledStakerShares);
+    }
+
+    /**
+     * @notice helper pure to calculate the shares to complete a withdrawal after slashing
+     * We use the totalMagnitude of the delegated operator at the time of withdrawal completion to scale the shares
+     * back to real StrategyManager/EigenPodManager shares
+     */
+    function _calculateSharesToCompleteWithdraw(
+        uint256 scaledStakerShares,
+        uint64 totalMagnitudeAtCompletion
+    ) internal pure returns (uint256 shares) {
+        shares = scaledStakerShares * totalMagnitudeAtCompletion / SlashingConstants.PRECISION_FACTOR;
+    }
+
+    /**
+     * @notice helper pure to return scaledShares given shares and current totalMagnitude. Used for
+     * adding/removing staker shares from operatorScaledShares
+     */
+    function _scaleShares(uint256 shares, uint64 totalMagnitude) internal pure returns (uint256) {
+        return shares * SlashingConstants.PRECISION_FACTOR / totalMagnitude;
+    }
+
+    /**
+     * @notice helper pure to return real strategy shares given operator scaledShares and current totalMagnitude.
+     * Used for returning the total delegated shares for an operator and strategy
+     */
+    function _descaleShares(uint256 scaledShares, uint64 totalMagnitude) internal pure returns (uint256) {
+        return scaledShares * totalMagnitude / SlashingConstants.PRECISION_FACTOR;
+    }
+
+    /**
+     * @notice staker scaling factor should be initialized and lower bounded to 1e18
+     */
+    function _getStakerScalingFactor(address staker, IStrategy strategy) internal view returns (uint256) {
+        uint256 currStakerScalingFactor = stakerScalingFactors[staker][strategy];
+        if (currStakerScalingFactor == 0) {
+            currStakerScalingFactor = SlashingConstants.PRECISION_FACTOR;
```

Comment:
overwrite storage here

Reply:
Didn't want to here since this function is used in `_getWithdrawableShares` as well